physical/oci_objectstorage/oci_objectstorage.go

```diff
@@ -0,0 +1,398 @@
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	response, err := o.client.GetObject(ctx, request)
+	o.logRequest("get", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
+
+	if err != nil {
+
+		if response.RawResponse != nil && response.RawResponse.StatusCode == http.StatusNotFound {
+			defer response.RawResponse.Body.Close()
+			return nil, nil
+		}
+
+		metrics.SetGauge(metricGetFailed, 1)
+		return nil, errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", key), err)
+	}
+
+	defer response.RawResponse.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Content)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		return nil, errwrap.Wrapf("failed to decode Value into bytes: {{err}}", err)
+	}
+
+	o.logger.Debug("physical/oci: get end")
+
+	return &physical.Entry{
+		Key:   key,
+		Value: body,
+	}, nil
+}
+
+func (o *Backend) Delete(ctx context.Context, key string) error {
+	o.logger.Debug("physical/oci: delete")
+	defer metrics.MeasureSince(metricDeleteFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricDeleteAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricDelete, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Debug("physical/oci: Delete: error generating UUID")
+		opcClientRequestId = ""
+	}
+	o.logger.Debug("physical/oci: Delete", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.DeleteObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.DeleteObject(ctx, request)
+	o.logRequest("delete", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+
+	if err != nil {
+
+		if resp.RawResponse != nil && resp.RawResponse.StatusCode == http.StatusNotFound {
+			defer resp.RawResponse.Body.Close()
+			return nil
+		}
+
+		metrics.SetGauge(metricDeleteFailed, 1)
+		return errwrap.Wrapf("failed to delete Key: {{err}}", err)
+	}
+	o.logger.Debug("physical/oci: delete end")
+	defer resp.RawResponse.Body.Close()
+
+	return nil
+}
+
+func (o *Backend) List(ctx context.Context, prefix string) ([]string, error) {
+	o.logger.Debug("physical/oci: list")
+	defer metrics.MeasureSince(metricListFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+
+	metrics.MeasureSince(metricListAcquirePool, startAcquirePool)
+	defer metrics.MeasureSince(metricList, time.Now())
+	var keys []string
+	delimiter := "/"
+	var start *string
+
+	for {
+
+		opcClientRequestId, err := uuid.GenerateUUID()
+		if err != nil {
+			o.logger.Debug("physical/oci: List: error generating UUID")
```

Comment:
I'd just return the `err` here. Not being able to get a random number is indicative of low level problems and we'd want to fail the request.